### DISCUSSION TITLE
Debugging a single plugin

### DISCUSSION THREAD

**Discussion by lnedry** [haraka/Haraka#3235] [General]
Is there a way to change the loglevel for a single plugin?

Maybe something similar to this so I wouldn't have to restart Haraka?
`echo DEBUG > config/loglevel
`

**Comment by msimerson** [comment]
I remember a conversation about this (can't remember if on email, slack, or here on GH) from a few years back. Short answer: not currently easy.

Within a plugin, you typically use a command such as this:

```js
    this.loginfo(this, "some log message");
```

And it's the 'loginfo' that is setting the log level of that message. The logging plugin then determines if the message should be logged based on the global logging level.

Within a plugin, you could probably hack up a solution that lets the plugin override the specified level based on a config setting. Completely untested and probably-not-working example:

```js
    this.log(this, 'info', "some log message")

    log (level, msg) {
        const le = this.cfg.loglevel ?? level;
        this[`log${le}`](this, msg)
    }
```
